runtime/alloc.c

```diff
@@ -316,7 +316,14 @@ CAMLprim value caml_update_dummy(value dummy, value newval)
 
   tag = Tag_val (newval);
 
-  if (tag == Double_array_tag){
+  if (Wosize_val(dummy) == 0) {
+      /* Size-0 blocks are statically-allocated atoms; we cannot
+         mutate them, but there is no need as the runtime only uses
+         dummies of tag 0 for valid OCaml values -- even, say, for
+         0-sized flat float arrays. So doing nothing suffices. */
```

Comment:
It's hard to write unambiguous sentences that use "only".  Suggested rewrite:
```suggestion
      /* Size-0 blocks are statically-allocated atoms. We cannot
         mutate them, but there is no need since the dummy was
         created with tag 0 and since all atoms used
         to represent OCaml values have tag 0, including empty
         arrays and empty flat float arrays. So, doing nothing suffices. */
```

Reply:
Thanks! I updated. the comment.